### PR TITLE
ci!: add and enforce Prettier formatting

### PR DESCRIPTION
**Pull request by redeboer**
Part 1 of #76  

Also updated the pre-commit hooks

### CODE REVIEW

File: doc/adr/0000-use-markdown-architectural-decision-records.md [hunk 1]
```diff
@@ -14,20 +14,20 @@ Which format and structure should these records follow?
...
-* [MADR](https://adr.github.io/madr/) 2.1.2 – The Markdown Architectural Decision Records
-* [Michael Nygard's template](http://thinkrelevance.com/blog/2011/11/15/documenting-architecture-decisions) – The first incarnation of the term "ADR"
-* [Sustainable Architectural Decisions](https://www.infoq.com/articles/sustainable-architectural-design-decisions) – The Y-Statements
-* Other templates listed at [github.com/joelparkerhenderson/architecture_decision_record](https://github.com/joelparkerhenderson/architecture_decision_record).
-* Formless – No conventions for file format and structure
+- [MADR](https://adr.github.io/madr/) 2.1.2 – The Markdown Architectural Decision Records
+- [Michael Nygard's template](http://thinkrelevance.com/blog/2011/11/15/documenting-architecture-decisions) – The first incarnation of the term "ADR"
+- [Sustainable Architectural Decisions](https://www.infoq.com/articles/sustainable-architectural-design-decisions) – The Y-Statements
+- Other templates listed at [github.com/joelparkerhenderson/architecture_decision_record](https://github.com/joelparkerhenderson/architecture_decision_record).
+- Formless – No conventions for file format and structure
```
Comment (spflueger) [inline doc/adr/0000-use-markdown-architectural-decision-records.md:21]: So `*` is not prettier compatible?

Comment (redeboer) [reply doc/adr/0000-use-markdown-architectural-decision-records.md:21]: Yeah I didn't do this, it's what happens after running `pre-commit run prettier -a`

**Comment by redeboer** [comment]
@spflueger Do we want a [Prettier badge](https://github.com/prettier/prettier-vscode#prettier-formatter-for-visual-studio-code)? 😏

**Comment by spflueger** [comment]
> @spflueger Do we want a [Prettier badge](https://github.com/prettier/prettier-vscode#prettier-formatter-for-visual-studio-code)?

yeah good point. We can add that

**Pull request merged**